### ISSUE TITLE
[BUG]: C++ CMake template / cmake_minimum_required() command should be the first command in CMakeLists.txt 

### ISSUE THREAD

**Issue by nikidimitrow** [compiler-explorer/compiler-explorer#5762]
### Describe the bug

By using the C++ CMake templated from godbolt.org generated the following warning:

```
CMake Warning (dev) at CMakeLists.txt:1 (project):

cmake_minimum_required() should be called prior to this top-level project() call.  Please see the cmake-commands(7) manual for usage documentation of both commands.

This warning is for project developers.  Use -Wno-dev to suppress it.
```

### Steps to reproduce

1. Click on "Templated" menu
![image](https://github.com/compiler-explorer/compiler-explorer/assets/10328871/272dd27e-91da-481a-9175-7ba97ad632b1)
2. Select C++ Cmake
![image](https://github.com/compiler-explorer/compiler-explorer/assets/10328871/3e7a5d63-572f-4881-90f9-cbb4683fd1a1)
3. Once new templated is open, navigated to the CMakeList.txt
![image](https://github.com/compiler-explorer/compiler-explorer/assets/10328871/38c8c22d-c4f4-4e5a-b7a6-1af99e3a5270)

### Expected behavior

The cmake_minimum_required() command should be the first command in CMakeLists.txt file, followed by the project() command. This is because cmake_minimum_required() tells CMake which version of CMake is required to build your project, and project() sets the name of your project and other basic information.

```
cmake_minimum_required(VERSION 3.5)

project(shapes)
```

### Reproduction link

https://godbolt.org/#z:OYLghAFBqd5QCxAFwE4FN0BoCWIIDGAtgIYDW6AgqsAM4gDkAtACIDCAspQNICiA%2BgCEAqgEkAMi34AVAJoAFXgFIAzCwBK6ADYB1HMgQt0AI1EA7AGYB7BlgJWiABxxb0qcSTPAAriWDpRABMQAiUAJkFwwTtvWmQHAHlvZEdkgDEXdDMSInQUBHR%2BdAAPdAJkkmNXLAtM%2Blh0QP0rVCCQADYazLaARgAWLtds3JBSHDMAOgJHRyxQBlscWnMCLW9AxsYABkXaDhJxgGUrb1QCPIYe3YTHLO3bLU9gNoWASiwIRubW4J7BgN%2BAFZ/sM8pxyOhxEtkLQJshisg5oxHKgrAArMrICC0BAkW60V5KLaUImUYgQ/hEcY4IjeIj8DAAR28OAwgQgADVeOpDqIEgA5ADUKgmgMJxNJJECgSKpXKyEqrmxuPxzJIGFJgq1gpxePQtAA9ARWat0FMZprtbr8QaMAQFV5XObHJatWNJtNHOKScSFTR0Mh%2BFpxmQgzhjKh1Th9cq9bQ1RridrBRYiMhAt7dis1htggwdrg9gczMdTud7oWbnd8w8nm1yRR3p8mvEfiAwv9ep1akMcnkMLQTmd9QaClotFY4QikQxSQAJbQTrCCgDuLS0gTAYFJpKYe6YpOkBUFvDMgVJWbMq3Wmxrhf2RyH5cu11uZgrjy89aiUSbX1bbQqJ2vwAJwgn2IA4JOnozpe165hWSwPiWT4XFclZvh%2BdZ5gwf4ti0bQDD2ALtls4EjFBEwILBhbZjeeYFkhxalsOjDoUsVbvnen7PDhrzvEs4IUPIqIYvaFZoiAfxmOgK4ZK4bRhD07xaIwgK2FxBZWIw0gYOggrhCo6HIPc/FYGQpGin0ADsIF9ICYRbICNl2c5WCqQwfQaRW2kLFg9BkSZNZNjAUChRASD2BsvD4aguD4MRoKjMWzo1FYZjIIcBAkK4UlEelyDCLQ6DyMUEkdO5TwhD%2BET%2BdomJQe%2BnxnmwVhrEQ75/FkgRQjJ/J0sYbggAAHFgjhWLQ%2BiNa17WdWNE1TelvXoP1RCDagI11a49qNYcfrIDNdJzcV23ILt%2B3Lat62bXE6oHW1R1Sf5F3jCtA1DcN7yDmWFyGeM8H6aobCpsgRotGaCCqMoEoSmEKj/TmgNhGElGQ8jO7EuMyCCu6ECvAZ1mREm2ogyAIAoljEDhGE8n6fYGVZDCIBKICbBmEohMcyw1PLv4gaBCQCp44SKhEyS1nc8SuHuWp3l3tp7qpUFOxNpFVjRbF8UQIlEFCZC0KwvCiLWBlWU5Xk/RpRlRUlWVd6SX8PEhKQFBbQ16X4N1h0dSA1lYN1l3vRtnTjZNZ3pd775%2B6Hi1mIHa1DZ0J3uyW%2B2R77bs7ele13fH11J2nD0%2B37t2oMgeeJ19qHIqJmKxvi3qkg2hRUmYNJ0gy6DMqyjSctyvICsKoqN8SUoyiUZQVFU6AQAYhQT/KiroK6gqUZ6K%2BK56I%2BUH6fNhmYobBhGUYxnPsqTwq08ryDGYXipstYJptjaXrUJxIb07K9LEUhOr6Axd8LWOsRgDlQoaMcE4pzGwKmbXKlsTaFWKqVcqHYnahAiFEfy1dZzEgXOOKwy41yoA3FuDGlB9wHmJEefSp5zxS3vgwdSj8fIgFAT9cBi5JxGywF/VWv8NaALwNrTISU14zCtplbKcD8rWyQXbAskkgJoJqtEZOWcmpeyLp1ICAdXpXUTvNMO00tFSSAjHcOcc9FBwqmoixOcy7px6EBWx51c5WITsHZ6d1HHOJen1ax7Qq4/UYH9K8iMDIqGBnEDAOQoY7jhgjG8ES2DRPGMAOJMNKBxGCCAVJXhBR7wFkLfGHMxbJmyWTHAFhonoByKvapVNkZsOHBw/BUDqYizKdqWIaTBSiGxqoFgOp0yVJqTkYw3gLD8H0G4QWLQgYEFxKgOJosV4YGQKcMwwycl5OABAfpEAqkEmXAc/iqgxZczvjLRhcstKQWguI3hHw1YCNbEAkREFUYSNgRbGRiDbblQGMojBtUXEe2aoERx/szwVw2tHBaFioXmMarCjOYLU7eJMX8dF9jy7uOuiXQus0nqlzxf4jxvsgmsVnAkiwZ50AWEFGkBICR%2BBzniSoDYtQZJMpZWyshoSAbJN2Rkn0WSRm5LQL0wpgsSB43OQKuG3UqlXI8kwp%2BWBtJfN4e8AoUohp43MiAPow0Jh9HNRay1lrrleSwEQEAgIyIat8rYAKPDTIfHCuFNWThMhxSEfYX1rgNrAB6FsICPZkBuHoBAYwFZjDjHVAATzYrYQNuQMoJDMFoFNd4NikEdBcAsTQ7RnQAG76grAvZIaFFgMw8gWI%2BkZUBJtahsCsUr7UvneBYR4dAOTRhXJxVN3ypG/IkTbZB9sQBKKqugyItUj71CgElEGWAK0bQYNZJxuF3hWEcBY%2BgzBsmDPnVERygomAJBUJe3t8RXX1XUZ7FqWLoU9XxUNP4yKI6vu/ZY8l11sWPrsUSx6QHTquLLqi7FoGfYwbcQBz9VLnzGT0lJBh6qWEMGKMNdoTB2h9AKQQAggow0TBvRANgKjly6UwAZOGfxBStSDW4ejKgVCvFsDqrAeqNgbUNRZQEIpnLtC2CBECYQQJbGssNLYTigWMFtfavoWwrLDRUCp5y6nNOdGdYwV1IBAoergLAJAJwUjJCATAsdeUJ1yPKkBNAmAcC/CwCuSMjhU0Yduc/RgSQLPYysIynDeGCNEZI2RijzHnDBrYxxrjpldW1L4/gBhSn2ztAmD0Ho6mwjtGy7l/LHY9N%2BTdU8kziAQDVreUI9UdB5CeG0IcBAVgVwasDTFwWjUGsyS0M11rXF0Idb9SAUNPQOzDeDf5/rbWR0L2oHQVh%2BowGjk4VA/2co9pNC4rYBBPzbMIMnfI2wDtKpeGqiC6IS74AQFXWmddQ0t07rMvuw9jAmAnrUGeiIF6r03qYHemwWDgkvmettnrTWWuzdB05i2bmPNeeuZh%2BWjBeBymSC0QUIX8OEeAMR0jPQJhhEFJR6jgpaNIyMpx91wU0u2Htex8j7Hmcs5Z8wlHpXDM05Vu8QThPAQC8F0LwXnQPIqB85q/T3Pd1GtsmahyTkXL2UcoCUXjAwj06klsJ1WGEu0%2BuRr9ndy/LcY3ZND2fQgA%3D%3D%3D

### Screenshots

![image](https://github.com/compiler-explorer/compiler-explorer/assets/10328871/2f61a5ee-64a0-498a-aaed-9cfb642040b9)

### Operating System

Windows 10

### Browser version

Google Chrome Version 119.0.6045.125 (Official Build) (64-bit)
